CONTRIBUTING.md

```diff
@@ -0,0 +1,287 @@
+>
+> In other scenarios, multiple types could be appropriate because of the nature of the
+> commit. This can happen with test and docs, which can be used as either types or
+> scopes.
+>
+> Run down the following list and select the highest-ranked type that fits your change:
+>
+> - ci
+> - build
+> - feat
+> - fix
+> - perf
+> - refactor
+> - style
+> - test
+> - docs
+> - chore
+
+Committing triggers the pre-commit hook, which runs the automatic code formatter and the
+fast linters.
+
+If the linters reformatted any of the files, the commit was cancelled. To make the
+changes stick, restage the modified files with `git add -A` and commit again.
+
+### Test the change
+
+Test early and often, especially before you plan to open a pull request.
+
+For low-complexity changes that require basic unit testing, run the fast tests:
+
+```bash
+make test-fast
+```
+
+For complex work, run the full test suite:
+
+```bash
+make test
+```
+
+Running all tests can take a very long time, in some cases an hour.
+
+When iterating and testing, it's a good practice to clean the local temporary files that
+the tests generate:
+
+```bash
+make clean
+```
+
+In rare instances, tests can fail in unpredictable ways, regardless of the state of your
+code. In such cases, it's best to delete your virtual environment and start over:
+
+```bash
+rm -rf ./.venv
+make clean
+make setup
+```
+
+There are also special tests that are feature-specific. Run `make help` to view all of
+them.
+
+### Document the change
+
+Craft CLI has frequent documentation updates to both its user guide and its developer
+material.
+
+Document feature changes and fixes in the [changelog](docs/changelog.rst).
+
+For all documentation updates, build the site locally with:
+
+```bash
+make docs
+```
+
+Use the special documentation linter to check for errors:
+
+```bash
+make lint-docs
+```
+
+### Push the branch and open a PR
+
+Once your work is committed to your branch, push it to your fork:
+
+```bash
+git push -u origin <branch-name>
+```
+
+Finally, [open a PR](https://github.com/canonical/craft-cli/compare) for it on GitHub.
+If your branch has one commit, GitHub will title the PR after it. If your branch has
+more than one commit, name the PR after the most significant. Once open, reviewers are
+assigned automatically to your work.
+
+### Follow up for the review
+
+When a maintainer reviews your PR, they typically leave inline comments and suggestions
+on the code.
+
+If the comment is a request, accommodate it by pushing one or more additional commits to
+the branch. It's simplest to add the commits locally and push, rather than in the GitHub
+interface, as it leads to fewer potential conflicts with syncs.
+
+Don't force-push changes to the branch. It destroys the history of the review and makes
+it harder for maintainers to see code changes.
+
+### Evaluating pull requests
+
+Oftentimes all you want to do is see if a given pull request solves the issue you were
```

Comment:
this section as a whole doesn't really apply to libraries (only to applications). We don't publish snaps for libraries

Reply:
Oops! That wasn't supposed to make it in. 1bd7129